addons/mail/static/tests/owl/component/chat_window_manager_tests.js

```diff
@@ -601,6 +601,423 @@ QUnit.test('chat window: open / close', async function (assert) {
+                        message_type: 'comment',
+                        model: 'mail.channel',
+                        record_name: 'General',
+                        res_id: i + 1,
+                    }
+                });
+            }
+            return this._super(...arguments);
+        },
+    });
+
+    // Open thread
+    document
+        .querySelector(`.o_MessagingMenu_toggler`)
+        .click()
+    ;
+    await testUtils.nextTick(); // re-render
+    document
+        .querySelector(`
+            .o_MessagingMenu_dropdownMenu
+            .o_ThreadPreviewList_preview`)
+        .click();
+    await testUtils.nextTick(); // re-render
+
+    // SETUP INITIAL STATE
+    // - Set a scroll position to chat window
+    document.querySelector(`.o_MessageList.o_Thread_messageList`).scrollTop = 142;
+
+    // - Set html content of the composer of the chat window
+    let composerTextInput = document.querySelector(`
+        .o_ComposerTextInput
+        > .note-editor
+        > .note-editing-area
+        > .note-editable`);
+    composerTextInput.innerHTML = "XDU for the win !";
+    composerTextInput.dispatchEvent(new window.KeyboardEvent('input'));
+
+    // - Set attachments of the composer
+    const composerLocalId = Object.keys(self.store.state.composers)[0];
+    const composerAttachmentLocalIds = [
+        this.store.dispatch('createAttachment', {
+            filename: 'blah.png',
+            id: 1,
+            name: 'blah',
+        }),
+        this.store.dispatch('createAttachment', {
+            filename: 'blah.pdf',
+            id: 2,
+            name: 'bluh',
+        }),
+    ];
+    this.store.dispatch('linkAttachmentToComposer', composerLocalId, composerAttachmentLocalIds[0]);
+    this.store.dispatch('linkAttachmentToComposer', composerLocalId, composerAttachmentLocalIds[1]);
+    await testUtils.nextTick();
+
+    // TESTS
+
+    // - phase 1 : on hiding menu
+    // -- Hide home menu
+    await this.widget.call('chat_window', 'test:will_hide_home_menu');
+    await this.widget.call('chat_window', 'test:hide_home_menu');
+    let domAttachments;
+
+    // -- Check dom state
+    assert.strictEqual(document.querySelector(`.o_MessageList.o_Thread_messageList`).scrollTop, 142, "chat window scrollTop should still be the same");
+    composerTextInput = document.querySelector(`
+        .o_ComposerTextInput
+        > .note-editor
+        > .note-editing-area
+        > .note-editable`);
+    assert.strictEqual(composerTextInput.innerHTML, "XDU for the win !", "chat window composer should still have the same html input");
+    domAttachments = document.querySelectorAll(`.o_Composer .o_Attachment`);
+    assert.strictEqual(domAttachments.length, 2, "chat window composer should still have the same amount of attachments");
+    assert.strictEqual(domAttachments[0].dataset.attachmentLocalId, composerAttachmentLocalIds[0], "chat window composer should still have the same first attachment");
```

Comment:
Explicit querySelector + hard-coded value, should make this assertion much easier to read.
Should look like this:

```js
assert.strictEqual(
   document.querySelectorAll(`
      .o_Composer
      .o_Attachment[data-attachment-local-id="ir.attachment_1"]`)
   .length,
   1,
   "Chat window composer should still have attachment with ID 1");
```

(Even better, local ID should not leak in the test. I will add some store getters later, so that we can get local ID of attachment just by knowing its ID).